### PR TITLE
Add support for new plugin system

### PR DESCRIPTION
**Pull request by long1eu**
This introduces Web and macOS support.

Related to https://github.com/Lyokone/flutterlocation/issues/324

### CODE REVIEW

File: location_platform_interface/lib/src/method_channel_location.dart [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+/// The interface that implementations of `location` must extend.
+///
+/// Platform implementations should extend this class rather than implement it
+/// as `location` does not consider newly added methods to be breaking
+/// changes. Extending this class (using `extends`) ensures that the subclass
+/// will get the default implementation, while platform implementations that
+/// `implements` this interface will be broken by newly added
+/// [LocationPlatform] methods.
+class MethodChannelLocation extends LocationPlatform {
```
Comment (Lyokone) [inline location_platform_interface/lib/src/method_channel_location.dart]: Not sure why this is here. I've seen that `url_launcher` implements it that way, but I feel that the implementation should be in the `location` folder. What do you think ?

File: location_web/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+This package is the endorsed implementation of `location` for the web platform since version `3.0.0`, so it gets automatically added to your dependencies by depending on `location: ^3.0.0`.
+
+No modifications to your `pubspec.yaml` should be required in a recent enough version of Flutter (`>=1.12.13+hotfix.4`):
+
+```yaml
+...
+dependencies:
+  ...
+  firebase_core: ^3.0.0
```
Comment (Lyokone) [inline location_web/README.md]: Should be location

**Comment by long1eu** [comment]
I'm almost done with macOS. :D

File: location_platform_interface/lib/src/method_channel_location.dart (hunk 1 above)
Comment (long1eu) [reply location_platform_interface/lib/src/method_channel_location.dart]: This is how the firebase plugins implements it.

[functions](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_functions/cloud_functions_platform_interface/lib/src/method_channel_cloud_functions.dart)
[firestore](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_firestore.dart)
[auth](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel_firebase_auth.dart)

Comment (long1eu) [reply location_platform_interface/lib/src/method_channel_location.dart]: flutter.dev/go/federated-plugins search for `Code structure of a federated plugin`

Comment (Lyokone) [reply location_platform_interface/lib/src/method_channel_location.dart]: Really interesting thanks !

File: location/ios/Classes/LocationPlugin.m [hunk 3]
```diff
@@ -0,0 +1 @@
+/Users/long1eu/IdeaProjects/flutterlocation/location/darwin/LocationPlugin.m
```
Comment (Lyokone) [inline location/ios/Classes/LocationPlugin.m]: Shouldn't it be relative path ?

File: location/macos/Classes/LocationPlugin.m [hunk 4]
```diff
@@ -0,0 +1 @@
+/Users/long1eu/IdeaProjects/flutterlocation/location/darwin/LocationPlugin.m
```
Comment (Lyokone) [inline location/macos/Classes/LocationPlugin.m]: Same here for relative path

File: location/README.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+If your project was created before Flutter 1.12, you may need to follow [this](https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects).
+
+### iOS
+And to use it in iOS, you have to add this permission in Info.plist :
+```xml
+NSLocationWhenInUseUsageDescription
+NSLocationAlwaysUsageDescription
+```
+
+### macOS
```
Comment (Lyokone) [inline location/README.md]: Should we add the Web instructions here ?

Comment (long1eu) [reply location/README.md]: For the web there are no instructions needed

File: location/ios/Classes/LocationPlugin.m (hunk 3 above)
Comment (long1eu) [reply location/ios/Classes/LocationPlugin.m]: true, it's a mistake

File: location/macos/Classes/LocationPlugin.m (hunk 4 above)
Comment (long1eu) [reply location/macos/Classes/LocationPlugin.m]: true, it's a mistake

**Comment by Lyokone** [comment]
Great I'll merge that and work on the CI :)

**Comment by j05u3** [comment]
I don't know if you have done it or plan to but: it would be great to state this in the README.md

**Pull request merged**